lib/signs_ui/config/state.ex

```diff
@@ -49,6 +48,11 @@ defmodule SignsUi.Config.State do
     GenServer.call(pid, {:update_chelsea_bridge_announcements, changes})
   end
 
+  @spec update_sign_groups(GenServer.server(), SignGroups.t()) :: {:ok, t()}
```

Comment:
Following up from our Slack thread:

None of the `update` functions are really documented, so can you add a little `@doc ""` to each of them?

In terms of how they work, there's two "kinds": replace state, and merge state.

The updates for Headways and Chelsea Bridge just take an entirely new state, and replace the old state with it. The Chelsea Bridge value is just a boolean... so that makes sense. For headways, I think it's probably not ideal, but that's how it is.

The `update_sign_config` function, in contrast, takes just *new updates* and takes care of merging into the state for you. This is good since a PIO might update a single sign or a few signs out of the hundreds, and so the frontend only needs to send those minimal values, and then on the backend the channel only needs to send the diff to the GenServer.

With GenServers, it's useful to think about what kind of data is being passed back and forth in the messages, and all else equal, keep it minimal if possible. So while `SignsUi.Config.State` knows the whole state, entities working with it might not have to.

In this branch, the expiration logic does have to fetch the entire sign group state from the `Config.State` GenServer, to know what's expired. However, it _doesn't_ necessarily need to send the whole thing back. In this case, we're not talking about a _ton_ of data, so it doesn't matter too much, but I think it's a good principle to follow.

And then when thinking about how the frontend will interact with the state here, _it_ will only be sending updates for a particular sign group at a time.

So I think it would be ideal if this was more of a "merge" function like with signs, and not a "replace" function like headways.

And one more thing, from that other ticket – I think in spec'ing out these tickets I expected that other one to be done first, so sorry for not mentioning it in this one – the frontend can _delete_ a sign group by sending the unix timestamp key and an empty map.

Can you try changing this function a bit, then, to merge data like `%{"Red" => %{"1234" => %{}, "5678" => %{....}}` into the state, in this case, deleting `"1234"` from the state, and adding `"5678"` to it? And then with that behavior, the expiration function would be building up a map of routes/timestamps of empty maps, signalling `Config.State` to delete them.